packages/manifold/contracts/gachaclaims/IGachaLazyClaim.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+/**
+ * Gacha Lazy Claim interface
+ */
+interface IGachaLazyClaim {
+    enum StorageProtocol { INVALID, NONE, ARWEAVE, IPFS }
+
+    error InvalidSignature();
+    error FailedToTransfer();
+    error InsufficientPayment();
+    error PaymentNotAllowed();
+
+    event GachaClaimInitialized(address indexed creatorContract, uint256 indexed instanceId, address initializer);
+    // mvp: Gacha claims cannot be updated except for price
+    event GachaClaimUpdated(address indexed creatorContract, uint256 indexed instanceId);
+    event GachaClaimMintReserved(address indexed creatorContract, uint256 indexed instanceId, address indexed collector, uint256 amount);
+    event GachaClaimMintDelivered(address indexed creatorContract, uint256 indexed instanceId, address indexed collector, uint32 itemIndex, uint256 amount);
+    // nice to have:
+    // event GachaMintBatch(address indexed creatorContract, uint256 indexed instanceId, uint16 mintCount);
+
+    struct Recipient {
+        address receiver;
+        uint256 amount;
+        uint256 payment;
+        uint32 itemIndex;
+    }
+
+    struct Mint {
+        address creatorContractAddress;
+        uint256 instanceId;
+        Recipient[] recipients;
+    }
+        
+    struct ItemProbability {
+        uint32 itemIndex; // this is also the same as arweave index
+        uint16 rate; // given in basis pts
+        string tier;
+    }
+
+    struct ClaimParameters {
+        uint32 totalMax;
+        uint48 startDate;
+        uint48 endDate;
+        StorageProtocol storageProtocol;
+        string location;
+        address payable paymentReceiver;
+        address erc20;
+        uint256 cost;
+        uint256 startingTokenId;
```

Comment:
i dont think we'd have this (it'll be calculated through the implementation of 1155 gacha claim i think..?